### PR TITLE
[docs] Update recommended structure to remove Dagster code from `__init__.py`

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR updates the recommended folder structure to include the Definitions object in `definitions.py` instead of `__init__.py`. It also updates related guides and references.

Tutorials in the docs are not yet updated. Will open more PRs:
- Update the project in examples/
- Update the tutorials in examples/
- Update tutorials in docs
- Update Dag Uni content
- Update Dag Uni starter projects once we have a PR for Dag Uni content

## How I Tested These Changes

make mdx-full-format

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#23316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (maximearmstrong) [inline docs/content/guides/dagster/recommended-project-structure.mdx:135]: URL to be updated [here](https://github.com/dagster-io/dagster/pull/23350) after changes to project-fully-features are merged [here](https://github.com/dagster-io/dagster/pull/23345).

File: docs/next/components/mdx/includes/dagster/DagsterDevTabs.mdx
Comment (erinkcochran87) [inline docs/next/components/mdx/includes/dagster/DagsterDevTabs.mdx:23]:
```suggestion
Dagster can also load Python modules as [code locations](/concepts/code-locations). When this approach is used, Dagster loads the definitions defined in the module passed to the command line.

We recommend defining a variable containing the <PyObject object="Definitions" /> object in a submodule named `definitions` inside the Python module. In practice, the submodule can be created by adding a file named `definitions.py` at the root level of the Python module.

As this style of development eliminates an entire class of Python import errors, we strongly recommend it for Dagster projects deployed to production.
```

Let's break this up a bit - the paragraph is pretty visually dense and there's a lot to take in.

**Review by erinkcochran87 — COMMENTED** [review]
Left two suggestions, but it's looking good!

File: docs/next/components/mdx/includes/dagster/DagsterDevTabs.mdx
Comment (maximearmstrong) [reply docs/next/components/mdx/includes/dagster/DagsterDevTabs.mdx]: Updated in [32844cf](https://github.com/dagster-io/dagster/pull/23316/commits/32844cfc77898e48eb467649e4206d343f619b4f)

File: docs/content/concepts/code-locations.mdx
Comment (cmpadden) [reply docs/content/concepts/code-locations.mdx]: Confirmed that this has been updated. 👍

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (cmpadden) [reply docs/content/guides/dagster/recommended-project-structure.mdx:135]: @maximearmstrong linked PR has been approved if you want to merge / update URL.

**Review by cmpadden — APPROVED** [review]

**Pull request merged**
